internal/frameworkprovider/resource_user.go

```diff
@@ -0,0 +1,164 @@
+package framework
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/hashicorp/terraform-provider-corner/internal/backend"
+)
+
+type resourceUserType struct{}
+
+func (r resourceUserType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Attributes: map[string]tfsdk.Attribute{
+			"email": {
+				Type:          types.StringType,
+				Required:      true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{tfsdk.RequiresReplace()},
+			},
+			"name": {
+				Type:     types.StringType,
+				Required: true,
+			},
+			"age": {
+				Type:     types.NumberType,
+				Required: true,
+			},
+			// included only for compatibility with SDKv2 test framework
+			"id": {
+				Type:     types.StringType,
+				Optional: true,
```

Comment:
Nit: Should this be `Computed` and removed from the test configuration if it is not used in CRUD functionality?

Reply:
Excellent idea, although if I do that then I have to handle the `Unknown` value, or get this error:

```
--- FAIL: TestAccResourceUser (11.89s)
    resource_user_test.go:13: Step 1/1 error: Error running apply: exit status 1
        
        Error: Value Conversion Error
        
          with framework_user.foo,
          on terraform_plugin_test.tf line 2, in resource "framework_user" "foo":
           2: resource "framework_user" "foo" {
        
        An unexpected error was encountered trying to build a value. This is always
        an error in the provider. Please report the following to the provider
        developer:
        
        unhandled unknown value
```

We need to improve the framework error messaging here (and with `unhandled null value`) - opened https://github.com/hashicorp/terraform-plugin-framework/issues/191.

Definitely worth following up and expanding the provider implementation once we've got the CI hooked up.